wrangler-core/src/main/java/io/cdap/wrangler/utils/ColumnConverter.java

```diff
@@ -305,4 +311,38 @@ private static BigDecimal setDecimalScale(BigDecimal decimal, Integer scale, Rou
         "Cannot set scale as '%s' for value '%s' when rounding-mode is '%s'", scale, decimal, roundingMode), e);
     }
   }
+
+  public static Schema getSchemaForType(String type, Integer scale) throws DirectiveParseException {
+    Schema typeSchema;
+    type = type.toUpperCase();
+    if (type.equals("DECIMAL")) {
+      // TODO make set-type support setting decimal precision
+      typeSchema = Schema.nullableOf(Schema.decimalOf(38, scale));
+    } else {
+      if (!SCHEMA_TYPE_MAP.containsKey(type)) {
+        throw new DirectiveParseException(String.format("'%s' is an unsupported type. " +
+          "Supported types are: int, short, long, double, decimal, boolean, string, bytes", type));
+      }
+      typeSchema = Schema.nullableOf(Schema.of(SCHEMA_TYPE_MAP.get(type)));
+    }
+    return typeSchema;
+  }
+
+  static {
+    Map<String, Schema.Type> schemaTypeMap = new HashMap<>();
+    schemaTypeMap.put("INTEGER", Schema.Type.INT);
```

Comment:
probably its a good idea to refactor String `Type` into constants. These are used across this file.

Reply:
Done